tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/main/function/ArgumentParserPositiveTest.java

```diff
@@ -0,0 +1,264 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.test.main.function;
+
+import org.ballerinalang.BLangProgramRunner;
+import org.ballerinalang.launcher.LauncherUtils;
+import org.ballerinalang.util.codegen.ProgramFile;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Paths;
+
+/**
+ * Test class for entry function argument parsing positive scenarios.
+ *
+ * @since 0.982.0
+ */
+public class ArgumentParserPositiveTest {
+
+    private static final String MAIN_FUNCTION_TEST_SRC_DIR = "src/test/resources/test-src/main.function";
+
+    private ProgramFile programFile;
+    private ByteArrayOutputStream tempOutStream = new ByteArrayOutputStream();
+    private PrintStream defaultOut;
+
+    @BeforeClass
+    public void setUp() {
+        defaultOut = System.out;
+    }
+
+    @Test(dataProvider = "mainFunctionArgsAndResult")
```

Comment:
Shall we add descriptions for all the test methods?

Reply:
Should we be adding descriptions where the test names are self explanatory, given that they would be a duplication of the test names themselves?